### DISCUSSION TITLE
V2中如何在组件外部拿到schema？

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#1217] [Q&A 2.x] [answered]
1.x中可以通过FieldState中的props拿到schema，也可以使用getFormSchema;
2.x中如何在组件外部拿到schema？

**Comment by janryWang** [comment, marked as answer]
```
import {Schema} from '@formily/react'

const schema = new Schema({
  type:"object"
})

<SchemaField schema={schema} />
 
```

这样使用，内部会基于外部传入的Schema实例来组装真正的Schema，注意需要等组件渲染完成之后才能读

**Reply by NexxLuo** [reply]
我主要是想在effects里获取到字段的schema，基于1.x的时候在schema挂了很多自定义属性，effects中基于schema做了很多逻辑处理。
或者2.x里面，schema上的自定义属性有办法映射到fieldState中去吗？

**Reply by janryWang** [reply]
自定义属性你挂在component-props上就行了
